tests/components/homepluscontrol/test_integration.py

```diff
@@ -0,0 +1,794 @@
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210",
+        text=plant_modules,
+    )
+
+    # Load the entry
+    hass.data[DOMAIN] = {}
+    hass.config.components.add(DOMAIN)
+    config_entry = mock_config_entry
+    await setup.async_setup_component(hass, "http", {})
+    assert hass.http.app
+    await config_entry.async_setup(hass)
+    await hass.async_block_till_done()
+
+    # The setup of the integration calls the API 3 times
+    assert aioclient_mock.call_count == 3
+
+    # Check the entities and devices
+    await entity_assertions(
+        hass,
+        num_exp_entities=5,
+        expected_entities={
+            "switch.dining_room_wall_outlet": True,
+            "switch.kitchen_wall_outlet": True,
+        },
+    )
+
+
+async def test_plant_topology_reduction_change(
+    hass,
+    aioclient_mock,
+    mock_config_entry,
+    plant_data,
+    plant_topology,
+    plant_modules,
+    plant_topology_reduced,
+):
+    """Test an entity leaving the plant topology."""
+
+    # Register the mock responses
+    aioclient_mock.get(
+        PLANT_URL,
+        text=plant_data,
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210" + PLANT_TOPOLOGY_RESOURCE,
+        text=plant_topology,
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210",
+        text=plant_modules,
+    )
+
+    # Load the entry
+    hass.data[DOMAIN] = {}
+    hass.config.components.add(DOMAIN)
+    config_entry = mock_config_entry
+    await setup.async_setup_component(hass, "http", {})
+    assert hass.http.app
+    await config_entry.async_setup(hass)
+    await hass.async_block_till_done()
+
+    # The setup of the integration calls the API 3 times
+    assert aioclient_mock.call_count == 3
+
+    # Check the entities and devices - 5 mock entities
+    await entity_assertions(
+        hass,
+        num_exp_entities=5,
+        expected_entities={
+            "switch.dining_room_wall_outlet": True,
+            "switch.kitchen_wall_outlet": True,
+        },
+    )
+
+    # Now we refresh the topology with one entity less
+    aioclient_mock.clear_requests()
+    # Register the mock responses
+    aioclient_mock.get(
+        PLANT_URL,
+        text=plant_data,
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210" + PLANT_TOPOLOGY_RESOURCE,
+        text=plant_topology_reduced,
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210",
+        text=plant_modules,
+    )
+
+    # Need to patch the API to ignore the refresh interval settings
+    with patch(
+        "homeassistant.components.homepluscontrol.api.HomePlusControlAsyncApi._should_check",
+        return_value=True,
+    ) as mock_check:
+        coordinator = hass.data["homepluscontrol"][
+            "homepluscontrol_entry_id_coordinator"
+        ]
+        await coordinator.async_refresh()
```

Comment:
Move time forward instead with `async_fire_time_changed`.

https://github.com/home-assistant/core/blob/3e82509263098fd8940ba086444e04e16404163e/tests/common.py#L364-L366

Reply:
Understood - I added these mock events into the test code to initiate the api checks.